File: app/hmi/lib/features/boot_self_check/domain/boot_self_check_item.dart
import 'package:lws_hmi/l10n/app_localizations.dart';

/// Fixed-order boot self-check items (Modbus alarm-information aligned).
/// Camera Comm is owned by the async IP-camera product session, not self-check.
enum BootSelfCheckItem {
  controllerComm,
  pumpComm,
  gunComm,
  motorDriverTemp,
  gunMotorTemp,
  protectionMirrorTemp,
  collimatorTemp,
  wireFeederComm;

  String labelFor(AppLocalizations l10n) {
    switch (this) {
      case BootSelfCheckItem.controllerComm:
        return l10n.bootSelfCheckControllerComm;
      case BootSelfCheckItem.pumpComm:
        return l10n.pumpStatusText;
      case BootSelfCheckItem.gunComm:
        return l10n.gunHeadCommunicationText;
      case BootSelfCheckItem.motorDriverTemp:
        return l10n.motorDriverTempLabel;
      case BootSelfCheckItem.gunMotorTemp:
        return l10n.motorTempLabel;
      case BootSelfCheckItem.protectionMirrorTemp:
        return l10n.protectiveMirrorTempLabel;
      case BootSelfCheckItem.collimatorTemp:
        return l10n.collimatorTempLabel;
      case BootSelfCheckItem.wireFeederComm:
        return l10n.wireFeedingMachineCommunicationText;
    }
  }

  bool get requiresControllerReady => true;
}

/// Terminal / in-progress status for one self-check row.
enum BootSelfCheckStatus {
  checking,
  pass,
  fail,
  skipped;

  String labelFor(AppLocalizations l10n) {
    switch (this) {
      case BootSelfCheckStatus.checking:
        return l10n.bootSelfCheckStatusChecking;
      case BootSelfCheckStatus.pass:
        return l10n.bootSelfCheckStatusPass;
      case BootSelfCheckStatus.fail:
        return l10n.bootSelfCheckStatusFail;
      case BootSelfCheckStatus.skipped:
        return l10n.bootSelfCheckStatusSkipped;
    }
  }

  bool get isTerminal =>
      this == BootSelfCheckStatus.pass ||
      this == BootSelfCheckStatus.fail ||
      this == BootSelfCheckStatus.skipped;
}

File: app/src/main/java/com/lasercyber/lws/ui/common/boot/BootSelfCheckItem.java
package com.lasercyber.lws.ui.common.boot;

import androidx.annotation.StringRes;

import com.lasercyber.lws.ui.R;

/**
 * Fixed-order boot self-check items aligned with Monitor → Alarm Information tiles.
 */
public enum BootSelfCheckItem {

    CONTROLLER_COMM(R.string.boot_self_check_controller_comm, ItemKind.MODBUS_CONTROLLER),
    PUMP_COMM(R.string.pump_status_text, ItemKind.MODBUS_COMM),
    GUN_COMM(R.string.gun_head_communication_text, ItemKind.MODBUS_COMM),
    MOTOR_DRIVER_TEMP(R.string.motor_driver_temperature_text, ItemKind.MODBUS_TEMPERATURE),
    GUN_MOTOR_TEMP(R.string.gun_motor_temp_text, ItemKind.MODBUS_TEMPERATURE),
    PROTECTION_MIRROR_TEMP(R.string.protective_mirror_temperature_text, ItemKind.MODBUS_TEMPERATURE),
    COLLIMATOR_TEMP(R.string.collimator_temperature_text, ItemKind.MODBUS_TEMPERATURE),
    WIRE_FEEDER_COMM(R.string.wire_feeding_machine_communication_text, ItemKind.MODBUS_COMM),
    CAMERA_COMM(R.string.camera_comm_status_text, ItemKind.CAMERA);

    @StringRes
    private final int labelResId;
    private final ItemKind kind;

    BootSelfCheckItem(@StringRes int labelResId, ItemKind kind) {
        this.labelResId = labelResId;
        this.kind = kind;
    }

    @StringRes
    public int getLabelResId() {
        return labelResId;
    }

    public ItemKind getKind() {
        return kind;
    }

    public boolean requiresControllerReady() {
        return kind != ItemKind.CAMERA;
    }

    enum ItemKind {
        MODBUS_CONTROLLER,
        MODBUS_COMM,
        MODBUS_TEMPERATURE,
        CAMERA
    }
}
